File: Graph/DFS.py

adj = [[] for i in range(1001)] #mang luu cac dinh
visited = [False]*1001
def inp():
    n,m = map(int,input("Nhap so dinh va so canh: ").split());
    for i in range(m):
        x,y= map(int,input().split());
        adj[x].append(y);
        adj[y].append(x);
def dfs(u):
    print(u," ");
    visited[u] = True;
    for i in adj[u]:
        if visited[i] == False:
            dfs(i);
if __name__=="__main__":
    inp()
    dfs(1)

File: Graph/DFS.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;
int n,m;
vector<vector<int>> danh_sach_ke(1001);
vector<bool> visited(1001,false);
void inp(){
    cin >> n >> m;
    for (int i = 0; i < m ; i++){
        int x,y ; cin >> x >> y;
        danh_sach_ke[x].push_back(y);
        danh_sach_ke[y].push_back(x);
    }
}
void dfs(int u ){
    cout << u << " "; //in dinh u
    visited[u] = true; // danh dau dinh u da duoc thanm
    for (int v : danh_sach_ke[u]){ //duyet cac dinh ke voi dinh u
        if(!visited[v]){// neu dinh v ke voi dinh u chua duoc tham
            dfs(v);  // tim kiem voi dinh v
        }
    }
}
int main(){
    inp();
    dfs(1);
}